### PR TITLE
DEVX-2354: Update demo to use FMC kafka-connect-datagen: beginner CCloud CLI

### PR DESCRIPTION
**Pull request by awalther28**
### Description 

https://confluentinc.atlassian.net/browse/DEVX-2354

_What behavior does this PR change, and why?_

Replaces the docker datagen connector with a managed cloud datagen connector. 

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

 - [x] Documentation 

 - [x] ccloud/beginner-cloud 

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._
Docs staging link: http://staging-docs-independent.confluent.io.s3-website-us-west-2.amazonaws.com/docs-platform/PR/215/6.0.1/tutorials/examples/ccloud/docs/beginner-cloud.html

 - [ ] Documentation 

 - [ ] ccloud/beginner-cloud

### CODE REVIEW

File: ccloud/beginner-cloud/start.sh [hunk 1]
```diff
@@ -271,7 +270,7 @@ ccloud kafka acl delete --allow --service-account $SERVICE_ACCOUNT_ID --operatio
 ##################################################
 
 TOPIC3="demo-topic-3"
-
+CONNECTOR="datagen_ccloud_pageviews"
```
Comment (rspurgeon) [inline ccloud/beginner-cloud/start.sh:272]: nit alert:  WDYT about coordinating the file name and the connector name?  The terms are ordered differently.

File: ccloud/docs/beginner-cloud.rst [hunk 2]
```diff
@@ -637,73 +634,38 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
+      Created connector datagen_ccloud_pageviews lcc-zno83
 
 #. Run the following command to verify connector is running:
 
    .. code-block:: bash
 
-      curl --silent http://localhost:8083/connectors/datagen-demo-topic-3/status | jq -r '.'
+      ccloud::wait_for_connector_up connectors/ccloud-datagen-pageviews.json 240 || exit 1
```
Comment (rspurgeon) [inline ccloud/docs/beginner-cloud.rst]: I'm not sure we'd want to have the `exit` conditional in the directions. If we do the user will lose their shell instance if the command doesn't succeed

Comment (awalther28) [reply ccloud/docs/beginner-cloud.rst]: I'm glad you pointed that out. I removed that bit in the instructions and corrected the path to the .json config in this [bc73545](https://github.com/confluentinc/examples/pull/887/commits/bc73545dabaa3e4b8b011366fd24b25b0ee43d22)

File: ccloud/beginner-cloud/start.sh (hunk 1 above)
Comment (awalther28) [reply ccloud/beginner-cloud/start.sh:272]: Hehe I like the "nit alert" callout, I have edited the instructions, start script, and the json config file name to all match the connector name.

**Review by ybyzek — COMMENTED** [review]
@awalther28 nice job getting this working so quickly!  Some comments:

1. `Run kafka-connect-datagen connector with wildcard ACLs` — consider renaming `kafka-connect-datagen` to match the connector’s name in CCloud (all mentions, in addition to section heading)
2. Show the contents of `ccloud-datagen-pageviews.json` file in the docs (using `literalinclude`) — it’s nice in its simplicity, and literal include lets people copy/paste.
3. The ACLs should change…they no longer need to be configured for the connect worker, so they can be scoped down to enable the connector to write to a topic

File: ccloud/beginner-cloud/cleanup.sh [hunk 3]
```diff
@@ -27,13 +27,21 @@ CLUSTER_NAME="demo-kafka-cluster"
...
-# - Delete the Kafka topics, Kafka cluster, environment, and the log files
+# - Delete the managed Kafka connector, Kafka topics, Kafka cluster, environment, and the log files
 ##################################################
 
-echo -e "\n# Cleanup: delete topics, kafka cluster, environment"
+echo -e "\n# Cleanup: delete connector, topics, kafka cluster, environment"
+if [[ ! -z "$CONNECTOR" ]]; then
+    echo "ccloud connector delete $CONNECTOR"
+    ccloud connector delete $CONNECTOR 1>/dev/null
+    sleep 5
```
Comment (ybyzek) [inline ccloud/beginner-cloud/cleanup.sh]: Is the sleep necessary?

Comment (awalther28) [reply ccloud/beginner-cloud/cleanup.sh]: I just ran through the demo twice without it and everything appears to be operating as expected. I'll remove the sleep.

**Comment by awalther28** [comment]
I believe I have addressed all your comments @ybyzek, the staging docs link should reflect those latest changes.

File: ccloud/beginner-cloud/cleanup.sh [hunk 4]
```diff
@@ -27,13 +27,20 @@ CLUSTER_NAME="demo-kafka-cluster"
...
+# - Delete the managed Kafka connector, Kafka topics, Kafka cluster, environment, and the log files
 ##################################################
 
-echo -e "\n# Cleanup: delete topics, kafka cluster, environment"
+echo -e "\n# Cleanup: delete connector, topics, kafka cluster, environment"
+if [[ ! -z "$CONNECTOR" ]]; then
+    echo "ccloud connector delete $CONNECTOR"
+    ccloud connector delete $CONNECTOR 1>/dev/null
+fi
+
```
Comment (ybyzek) [inline ccloud/beginner-cloud/cleanup.sh:43]: @awalther28 should the following line also be optimized to remove `connect-configs connect-offsets connect-status`?

File: ccloud/beginner-cloud/start.sh [hunk 5]
```diff
@@ -271,20 +270,14 @@ ccloud kafka acl delete --allow --service-account $SERVICE_ACCOUNT_ID --operatio
...
```
Comment (ybyzek) [inline ccloud/beginner-cloud/start.sh:272]: Maybe also update this comment from:

```
##################################################
# Run Connect and kafka-connect-datagen connector with permissions
# - Confluent Hub: https://www.confluent.io/hub/
##################################################		 
```

to something along the lines of

```
##################################################
# Run the fully-managed connector kafka-connect-datagen in Confluent Cloud
##################################################		 
```

File: ccloud/docs/beginner-cloud.rst [hunk 6]
```diff
@@ -637,73 +585,40 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
+      Created connector datagen_ccloud_pageviews lcc-zno83
 
 #. Run the following command to verify connector is running:
 
    .. code-block:: bash
 
-      curl --silent http://localhost:8083/connectors/datagen-demo-topic-3/status | jq -r '.'
+      ccloud::wait_for_connector_up datagen_ccloud_pageviews.json 300
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]: This is a library function which is great for automation (and the `start.sh` script) but for the docs it is more instructive to have the user learn the Confluent Cloud CLI command `ccloud connector <list>` (or `<describe>`) instead.

File: ccloud/beginner-cloud/start.sh [hunk 7]
```diff
@@ -296,53 +289,15 @@ echo "../../ccloud/ccloud-generate-cp-configs.sh $CLIENT_CONFIG &>/dev/null"
...
-curl --silent http://localhost:8083/connectors/$CONNECTOR/status
-STATE=$(curl --silent http://localhost:8083/connectors/$CONNECTOR/status | jq -r '.connector.state')
-#echo $STATE
-if [[ "$STATE" != "RUNNING" ]]; then
-  echo "ERROR: connector $CONNECTOR is not running.  Please troubleshoot the Docker logs."
-  exit $?
-fi
+echo -e "\n# Wait for connector to be up"
+echo "ccloud::wait_for_connector_up $CONNECTOR.json 300"
+ccloud::wait_for_connector_up $CONNECTOR.json 300 || exit 1
```
Comment (ybyzek) [inline ccloud/beginner-cloud/start.sh]: IMO it's fine to use the `ccloud` library function here because it's in the bash script.

File: ccloud/docs/beginner-cloud.rst [hunk 8]
```diff
@@ -637,73 +585,40 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
-
-   Your output should resemble:
+#. Create a managed connector in Confluent Cloud with the provided configurations mentioned above using the following commands:
 
    .. code-block:: text
 
-      Creating network "beginner-cloud_default" with the default driver
-      Creating connect-cloud ... done
+      source ../../utils/ccloud_library.sh 
+      ccloud::create_connector datagen_ccloud_pageviews.json
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]: This is a library function which is great for automation (and the `start.sh` script) but for the docs it is more instructive to have the user learn the Confluent Cloud CLI command `ccloud connector create` instead.

**Review by ybyzek — CHANGES_REQUESTED** [review]
@awalther28 This tutorial is to teach users the CLI, not the library.

- In the docs, switch from using `ccloud::` library functions to using Confluent Cloud CLI commands.  
- The start.sh script can still use the `ccloud::` library functions for automation

File: ccloud/beginner-cloud/cleanup.sh (hunk 4 above)
Comment (awalther28) [reply ccloud/beginner-cloud/cleanup.sh:43]: Correct, I have removed the unnecessary topic deletions.

File: ccloud/beginner-cloud/start.sh (hunk 5 above)
Comment (awalther28) [reply ccloud/beginner-cloud/start.sh:272]: Corrected.

**Comment by awalther28** [comment]
I have updated the docs to highlight how to use the CCloud CLI rather than the utils/ccloud_library script which should resolve the remainder of your comments @ybyzek . Open to different word smithing, especially when it comes to waiting for the connector to be in a "RUNNING" state. Link to exact steps [here](http://staging-docs-independent.confluent.io.s3-website-us-west-2.amazonaws.com/docs-platform/PR/215/6.0.1/tutorials/examples/ccloud/docs/beginner-cloud.html#run-a-fully-managed-datagen-ccloud-pageviews-connector)

File: ccloud/beginner-cloud/start.sh [hunk 9]
```diff
@@ -295,55 +287,20 @@ echo "../../ccloud/ccloud-generate-cp-configs.sh $CLIENT_CONFIG &>/dev/null"
 ../../ccloud/ccloud-generate-cp-configs.sh $CLIENT_CONFIG &>/dev/null
 echo "source delta_configs/env.delta"
 source delta_configs/env.delta
+sed -i '' 's/ESN5FSNDHOFFSUEV/$CLOUD_KEY/g' datagen_ccloud_pageviews.json
```
Comment (ybyzek) [inline ccloud/beginner-cloud/start.sh]: @awalther28 clever solution 🎉 !  Now the file renders in the docs with the sample credentials, but also works for the `start.sh` script.

cc: @rspurgeon for review, especially if we want to evaluate this approach for other examples as well

File: ccloud/docs/beginner-cloud.rst [hunk 10]
```diff
@@ -21,7 +21,7 @@ the following steps:
 -  `Run a Java producer without ACLs`_
 -  `Run a Java producer with ACLs`_
 -  `Run a Java producer with a prefixed ACL`_
--  `Run kafka-connect-datagen connector with wildcard ACLs`_
+-  `Run a fully managed datagen_ccloud_pageviews connector`_
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
-  `Run a fully managed Confluent Cloud connector`_
```

File: ccloud/docs/beginner-cloud.rst [hunk 11]
```diff
@@ -540,7 +537,7 @@ Run a Java producer with a prefixed ACL
    You should see two ``Deleted ACLs.`` messages.
 
 
-Run kafka-connect-datagen connector with wildcard ACLs
+Run a fully managed datagen_ccloud_pageviews connector
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
Run a fully managed Confluent Cloud connector
```

File: ccloud/docs/beginner-cloud.rst [hunk 12]
```diff
@@ -625,86 +576,43 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
-             "worker_id": "connect:8083"
-           }
-         ],
-         "type": "source"
-      }
+           ID     |           Name            |    Status    |  Type  | Trace
+      +-----------+---------------------------+--------------+--------+-------+
+        lcc-zno83 | datagen_ccloud_pageviews  | PROVISIONING | source |
 
+   If the ``Status`` is ``RUNNING`` you may move on to the next step.
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
   When the ``Status`` is ``RUNNING`` you may move on to the next step.
```

File: ccloud/docs/beginner-cloud.rst [hunk 13]
```diff
@@ -625,86 +576,43 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
-         )
-
-         curl --silent --output /dev/null -X POST -H "Content-Type: application/json" --data "${DATA}" http://localhost:8083/connectors
-
-
-#. Wait about 20 seconds for the kafka-connect-datagen connector to start producing messages.
+      Created connector datagen_ccloud_pageviews lcc-zno83
 
-#. Run the following command to verify connector is running:
+#. The connector may take up to 5 minutes to provision, use the following command to check if the
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
#. The connector may take up to 5 minutes to provision. Run the following command to check the connector status
```

File: ccloud/docs/beginner-cloud.rst [hunk 14]
```diff
@@ -625,86 +576,43 @@ Run kafka-connect-datagen connector with wildcard ACLs
...
+   connector is `PROVISIONING` or `RUNNING`.
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
```

File: ccloud/beginner-cloud/start.sh (hunk 9 above)
Comment (awalther28) [reply ccloud/beginner-cloud/start.sh]: One thing I felt was hacky with this route was that sed changes the config file so I added a `git checkout datagen_ccloud_pageviews.json` after the connector is up and running. This way if we are testing changes with start.sh, we won't accidentally add the changed config file. Open to other ideas on the rendering/injecting creds/cleanup/etc.

File: ccloud/docs/beginner-cloud.rst [hunk 15]
```diff
@@ -795,17 +701,38 @@ Run a Java consumer with a Wildcard ACL
...
+      Which should display a something similar to below. Locate your connector ID, in this case the connector ID is ``lcc-zno83``.
+
+      .. code-block:: text
+
+              ID     |           Name           | Status  |  Type  | Trace
+         +-----------+--------------------------+---------+--------+-------+
+           lcc-zno83 | datagen_ccloud_pageviews | RUNNING | source |
+
+
+   b. Delete the connector using the ID:
```
Comment (ybyzek) [inline ccloud/docs/beginner-cloud.rst]:
```suggestion
   b. Delete the connector, referencing the connector ID from the previous step:
```

File: ccloud/beginner-cloud/start.sh (hunk 9 above)
Comment (ybyzek) [reply ccloud/beginner-cloud/start.sh]: I see what you mean.  The alternative is to leave the json file with `$CLOUD_KEY` etc params, and tell the user to substitute their own credentials (which is the route other examples use).  Either way, the user needs to do a substitution...unless they are using env param substitution.

Comment (awalther28) [reply ccloud/beginner-cloud/start.sh]: @ybyzek Would the preferred route here be to leave the `$CLOUD_KEY`? Just to note, the user would need to manually set the env params, this PR suggests removing the following steps that used to set the env params. If we would prefer users leverage the generate-cp-configs I can add that step back. 
```
../../ccloud/ccloud-generate-cp-configs.sh /tmp/client.config &>/dev/null
source delta_configs/env.delta
```
Although I think manually substituting the key + secrets is more inline with other steps in this tutorial (see "Create a local configuration file /tmp/client.config..." [step 5](https://docs.confluent.io/platform/current/tutorials/examples/ccloud/docs/beginner-cloud.html#create-a-new-service-account-with-an-api-key-secret-pair)).

Comment (ybyzek) [reply ccloud/beginner-cloud/start.sh]: Another option to throw into the mix 😄 : to avoid the hacky `git checkout datagen_ccloud_pageviews.json`, the `start.sh` script can work off a _copy_ of the json connector file and do a substitution to use the temporary credentials.

Comment (awalther28) [reply ccloud/beginner-cloud/start.sh]: So many options, oh my! @rspurgeon do you have a preference here? 

A quick recap, we're trying to decide the best route to present a fully managed connector config file and how to use that config file in an automated stat script. 
The options appear to be: use env vars and add the process to set them in the tutorial instructions, create a unique config file for the start script and the tutorial rendering, or use the same config file for both and for the automation leverage `sed` to input secrets +`git checkout` to revert the config file. 

As stated earlier, I think the latter two better fit with the tutorial.

Comment (awalther28) [reply ccloud/beginner-cloud/start.sh]: @ybyzek I spoke with @rspurgeon offline. The conclusion we came to is to have the `start.sh` script work off of a copy of the json connector file housed in a directory that is ignored by bit. This is something similar to how kafka-devops works. The work has been validated by myself and pushed up to this branch.
 To my knowledge, this is the last standing issue for this PR.

Comment (ybyzek) [reply ccloud/beginner-cloud/start.sh]: @awalther28 Thanks for working through that, the approach looks good.

**Comment by ybyzek** [comment]
@awalther28    hopefully last comment (sorry for not bringing it up earlier but in light of our team conversation this week) Now that this PR creates a fully-managed connector which has hourly charges...a failure mid-script (or it going to RUNNING after 300s) could potentially leave the connector running accruing charges unbeknownst to the user.  As a result, can you please add the confirmation `prompt_continue_ccloud_demo` to the top of the start.sh script?  e.g. https://github.com/confluentinc/examples/blob/6.0.1-post/cloud-etl/start.sh#L34

File: ccloud/beginner-cloud/start.sh [hunk 16]
```diff
@@ -15,11 +15,11 @@ source ../../utils/ccloud_library.sh
 
 ccloud::validate_version_ccloud_cli 1.7.0 || exit 1
 ccloud::validate_logged_in_ccloud_cli || exit 1
+ccloud::prompt_continue_ccloud_demo || exit 1
```
Comment (ybyzek) [inline ccloud/beginner-cloud/start.sh:18]: @awalther28 approve, LGTM.  

One more note, if you're willing, I think we can remove `check_expect || exit 1` (line 21 below) but no worries if you want to leave that for another PR.

**Review by ybyzek — APPROVED** [review]

File: ccloud/beginner-cloud/start.sh (hunk 16 above)
Comment (awalther28) [reply ccloud/beginner-cloud/start.sh:18]: @ybyzek What does `check_expect` do?

Comment (awalther28) [reply ccloud/beginner-cloud/start.sh:18]: nvm

Comment (awalther28) [reply ccloud/beginner-cloud/start.sh:18]: Removed, looks like we don't use `expect` anymore in the `start.sh` script.

Comment (ybyzek) [reply ccloud/beginner-cloud/start.sh:18]: Before Confluent Cloud CLI supported `~/.netrc` for saved logins, the demos used `expect` to login & prompted user for username/password
